public/js/pimcore/helpers.js

```diff
@@ -3534,3 +3534,59 @@ pimcore.helpers.sendRequest = function (
     request.open(method, url);
     request.send();
 };
+
+pimcore.helpers.localizedDateTime = function (value, options){
+    const user = pimcore.globalmanager.get("user");
+    if (typeof options === 'undefined') {
+        options = {
+            dateStyle: "short",
+            timeStyle: "medium"
+        }
+    }
+
+    if (user.datetimeLocale) {
+        const dateFormatter = new Intl.DateTimeFormat(user.datetimeLocale, options);
+        return dateFormatter.format(value);
+    }else{
+        let format = '';
+        let separator = ' ';
+        if (options.dateStyle === 'short') {
+            format+= 'Y-m-d';
+        }
+        if (options.timeStyle === 'short') {
+            format+= separator + 'H:i';
+        }else if (options.timeStyle === 'medium') {
+            format+= separator + 'H:i:s';
+        }
+        if (format === ''){
+            format = 'Y-m-d H:i:s';
+        }
+        return Ext.Date.format(value, format);
+    }
+}
+
+pimcore.helpers.intlDateFormatFromLocale = function (fallback){
```

Comment:
this one doesn't work really well 😞 leading zero is missed

![image](https://github.com/pimcore/admin-ui-classic-bundle/assets/6014195/3477a519-3b99-4c3d-b10a-fdbca98d6a28)
![image](https://github.com/pimcore/admin-ui-classic-bundle/assets/6014195/9c8ae331-8370-473f-a653-54454ea95189)

Reply:
potentially solved by https://github.com/pimcore/admin-ui-classic-bundle/pull/554/commits/dfe9f77d6b03ef7721292423ff8e3793bffc17a2 but not sure if it's a good approach.

To avoid using any extra external library and to avoid to save hardcoded formats in some multi dimensional array in the js

Reply:
For time https://github.com/pimcore/admin-ui-classic-bundle/pull/554/commits/c5648d84c3a1d359df8103969b410eb9696e627d
![image](https://github.com/pimcore/admin-ui-classic-bundle/assets/6014195/5347c71b-dac0-4ea2-8014-7fbf9658c254)